pkg/api/validation/validation.go

```diff
@@ -522,6 +527,22 @@ func validateGlusterfs(glusterfs *api.GlusterfsVolumeSource) errs.ValidationErro
 	return allErrs
 }
 
+func validateFlocker(flocker *api.FlockerVolumeSource) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if flocker.DatasetName == "" {
+		allErrs = append(allErrs, errs.NewFieldRequired("datasetName"))
+	}
+	if strings.Contains(flocker.DatasetName, "/") {
+		allErrs = append(allErrs, errs.NewFieldInvalid("datasetName", flocker.DatasetName, "must not contain '/'"))
+	}
+	if flocker.Size != "" {
+		if _, err := units.FromHumanSize(flocker.Size); err != nil {
```

Comment:
If this is to go in for v1.1 it needs to be approved and clear of
encumberances by EOD Friday.  Otherwise it's going to miss the boat.  Note
that this doesn't mean every nit is fixed, but that major structural issues
are resolved and small details will be fixed soon after in a followup.

On Wed, Sep 23, 2015 at 8:05 AM, Alexandre González <
notifications@github.com> wrote:

> In pkg/api/validation/validation.go
> <https://github.com/kubernetes/kubernetes/pull/14328#discussion_r40214231>
> :
>
> > @@ -522,6 +527,22 @@ func validateGlusterfs(glusterfs *api.GlusterfsVolumeSource) errs.ValidationErro
> >  	return allErrs
> >  }
> >
> > +func validateFlocker(flocker *api.FlockerVolumeSource) errs.ValidationErrorList {
> > +	allErrs := errs.ValidationErrorList{}
> > +	if flocker.DatasetName == "" {
> > +		allErrs = append(allErrs, errs.NewFieldRequired("datasetName"))
> > +	}
> > +	if strings.Contains(flocker.DatasetName, "/") {
> > +		allErrs = append(allErrs, errs.NewFieldInvalid("datasetName", flocker.DatasetName, "must not contain '/'"))
> > +	}
> > +	if flocker.Size != "" {
> > +		if _, err := units.FromHumanSize(flocker.Size); err != nil {
>
> Oh cool, I am going to take a look to that one. Sorry, I didn't know it
> exists and I checked the Godeps to see what you had there.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14328/files#r40214231>.
>